Unity(JUMBO) build support

Say-Y opened the pull request:
I'm familiar with Jumbo build, so I think it would be nice to have this feature integrated.

https://github.com/ubisoft/Sharpmake/pull/305
Apply modifications

jspelletier commented on Sharpmake/Project.Configuration.cs:
```diff
@@ -1605,6 +1605,9 @@ public string FastBuildUnityPath
             /// </remarks>
             public bool DoNotGenerateFastBuild = false;
 
+            // Unity builds support
```
```suggestion
            // Jumbo builds support
```

jspelletier commented:
Could modify one of the existing sample to add jumbo build support in one of the configuration at least?

Say-Y reviewed (commented):
Unity -> Jumbo

jspelletier commented on samples/JumboBuild/reference/projects/jumbobuild_vs2019_win32.vcxproj:
```diff
@@ -0,0 +1,248 @@
+    <PlatformToolset>v142</PlatformToolset>
+    <EnableUnitySupport>true</EnableUnitySupport>
+    <MaxFilesInUnityFile>0</MaxFilesInUnityFile>
+    <MinFilesInUnityFile>2</MinFilesInUnityFile>
```
You don't have 2 .cpp files in the project so jumbo build won't be used, I think.

sylvainfortin commented:
Hi @Say-Y. Since you are adding a new sample, could you add it to `SamplesDef.json`. Once added to this file, it will be automatically be added to CI along other samples. See https://github.com/ubisoft/Sharpmake/blob/main/docs/Samples.md.

Say-Y commented:
Hi @jspelletier , I added a few header and cpp files. (test1.h, test1.cpp, test2.h, test2.cpp)

Hi @sylvainfortin , 'JumboBuild' was added by referring to the 'HelloWorld' part of the 'SamplesDef.json' file. I don't need to worry about 'UpdateSamplesOutput.bat'?

Say-Y commented:
> > Hi @jspelletier , I added a few header and cpp files. (test1.h, test1.cpp, test2.h, test2.cpp)
> > Hi @sylvainfortin , 'JumboBuild' was added by referring to the 'HelloWorld' part of the 'SamplesDef.json' file. also removed JumboBuild's 'reference' folder. And I don't need to worry about 'UpdateSamplesOutput.bat'?
> 
> UpdateSamplesOutput.bat is used to generate the reference files

Should I also edit that file?

if not "%ERRORLEVEL_BACKUP%" == "0" goto error
call :UpdateRef samples JumboBuild                  JumboBuild.sharpmake.cs                    reference         JumboBuild

jspelletier commented on Sharpmake/Project.Configuration.cs:
```diff
@@ -1605,6 +1605,11 @@ public string FastBuildUnityPath
             /// </remarks>
             public bool DoNotGenerateFastBuild = false;
 
+            // Jumbo builds support
```
```suggestion
            // Jumbo builds support for msbuild
```

jspelletier commented on Sharpmake/Options.Vc.cs:
```diff
@@ -1348,6 +1348,20 @@ public enum EnableAsan
                 }
+
+                /// <summary>
+                /// Enable Jumbo/Unity builds
```
```suggestion
                /// Enable Jumbo/Unity builds for msbuild. Only usable with msbuild.
```

jspelletier commented on regression_test.py:
```diff
@@ -147,6 +147,7 @@ def launch_tests():
             Test("CSharpHelloWorld", "HelloWorld.sharpmake.cs"),
             Test("HelloWorld", "HelloWorld.sharpmake.cs"),
+            Test("JumboBuild", "JumboBuild.sharpmake.cs"),
             Test("HelloLinux", "HelloLinux.Main.sharpmake.cs"),
```
you need to modify update UpdateSamplesOutput.bat as well to be able update the reference files.

jspelletier commented on samples/JumboBuild/JumboBuild.sharpmake.cs:
```diff
@@ -0,0 +1,73 @@
+            Name = "JumboBuild";
+
+            AddTargets(new Target(
+                    Platform.win32 | Platform.win64,
```
please remove win32. it is useless for this sample purpose. Also, only keep Optimization.Release.
This will slowdown the ci for nothing.

jspelletier commented on SamplesDef.json:
```diff
@@ -209,6 +209,22 @@
+            "Commands":
+            [
+                "./RunSharpmake.ps1 -workingDirectory {testFolder} -sharpmakeFile \"JumboBuild.sharpmake.cs\" -framework {framework}",
+                "./Compile.ps1 -slnOrPrjFile \"jumbobuild_vs2019_win32.sln\" -configuration {configuration} -platform \"Win32\" -WorkingDirectory \"{testFolder}/projects\" -VsVersion {os} -compiler MsBuild",
```
Remove 32 bit here as well. Only keep release as well.

Say-Y commented:
I have applied your request.
thank you

jspelletier commented:
I ran the CI.
CI failed because you need to run the .bat to create the reference files.

Say-Y commented:
I have confirmed that UpdateSamplesOutput.bat fails without the following settings

```
AddTargets(new Target(
        Platform.win32 | Platform.win64,
        DevEnv.vs2019,
        Optimization.Debug | Optimization.Release
));
```

After the change, I confirmed that regression_test.py passed.
